tests/cra.test.js

```diff
@@ -31,86 +32,83 @@ test.after(async (t) => {
 
 test('homepage', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/`).then((res) => res.text())
+  const response = await got(`${url}/`).text()
 
-  t.regex(response, /Web site created using create-react-app/)
+  t.true(response.includes('Web site created using create-react-app'))
 })
 
 test('static/js/bundle.js', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/static/js/bundle.js`)
-  const body = await response.text()
+  const { body, statusCode, headers } = await got(`${url}/static/js/bundle.js`)
 
-  t.is(response.status, 200)
+  t.is(statusCode, 200)
   t.true(body.length > BUNDLE_MIN_LENGTH)
-  t.truthy(response.headers.get('content-type').startsWith('application/javascript'))
-  t.regex(body, /webpackBootstrap/)
+  t.true(headers['content-type'].startsWith('application/javascript'))
+  t.true(body.includes('webpackBootstrap'))
 })
 
 const BUNDLE_MIN_LENGTH = 1e2
 
 test('static file under public/', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/test.html`)
-  const body = await response.text()
+  const { body, statusCode, headers } = await got(`${url}/test.html`)
 
-  t.is(response.status, 200)
-  t.truthy(response.headers.get('content-type').startsWith('text/html'))
+  t.is(statusCode, 200)
+  t.true(headers['content-type'].startsWith('text/html'))
   t.true(body.includes('<h1>Test content</h1>'))
 })
 
 test('redirect test', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/something`, { redirect: 'manual' })
+  const { body, statusCode, headers } = await got(`${url}/something`, { followRedirect: false })
 
-  t.is(response.status, 301)
-  t.is(response.headers.get('location'), `${url}/otherthing.html`)
-  t.is(await response.text(), 'Redirecting to /otherthing.html')
+  t.is(statusCode, 301)
+  t.is(headers.location, `/otherthing.html`)
+  t.is(body, 'Redirecting to /otherthing.html')
 })
 
 test('normal rewrite', async (t) => {
   const { url } = t.context.server
+  // TODO: replace with got like the rest of the tests
```

Comment:
This is the only test that fails when switching to `got` - it throws with a status code of 404.
I've tested with `fetch` and `curl` and it passes.

I'm not sure what is the issue (yet) but don't think this should block the PR.